src/learning/pca.rs

```diff
@@ -0,0 +1,156 @@
+//! Principal Component Analysis Module
+//!
+//! Contains implementation of PCA.
+//!
+//! # Examples
+//!
+//! ```
```

Comment:
I think this example will be a little confusing for people who are not very familiar with PCA. In particular due to some api confusion that I'll bring up in another comment.

I think it could be improved  by explaining a little more what the prediction part does. And moreover what the assertion means, what is this value that we are comparing to `-0.6686`?